numba/targets/arraymath.py

```diff
@@ -2867,6 +2869,26 @@ def np_imag_impl(a):
 #----------------------------------------------------------------------------
 # Misc functions
 
+@overload(np.count_nonzero)
+def np_count_nonzero(arr, axis=None):
+    if not type_can_asarray(arr):
+        raise errors.TypingError("The argument to np.count_nonzero must be array-like")
+    
+    if (NumpyVersion(np.__version__) < '1.12.0') and (axis in (None, types.none)):
+        warnings.warn("axis is not supported on Numpy versions < '1.12.0'")
```

Comment:
Should I also test this warning?

Reply:
I think compilation should just abort, else we're just replicating a NumPy bug?

Reply:
Hum...makes sense. I will change the code to reflect this.